compose/cli/log_printer.py

```diff
@@ -64,28 +71,30 @@ def get_log_generator(container):
     return build_no_log_generator
 
 
-def build_no_log_generator(container, prefix, color_func):
+def build_no_log_generator(container, prefix, color_func, log_args):
     """Return a generator that prints a warning about logs and waits for
     container to exit.
     """
     yield "{} WARNING: no logs are available with the '{}' log driver\n".format(
         prefix,
         container.log_driver)
-    yield color_func(wait_on_exit(container))
+    if log_args['follow']:
```

Comment:
I think this should be `log_args.get('follow')` in case the follow key doesn't exist.